### PR TITLE
Mainnet dockers

### PR DESCRIPTION
**Pull request by someanon**
~Still has issues, will fix later~. Fixed.

### CODE REVIEW

File: docker/mainnet/.env.empty [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# This file contains list of environment variables required to deploy
+# testnet connector dockers. You should copy this file to `.env` and
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/.env.empty]: mainnet?

File: docker/mainnet/bitcoin-neutrino/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    echo "Setting external IP"
+    EXTERNAL_IP_OPT="--externalip=$EXTERNAL_IP"
+fi
+
+RPC_USER_OPT="--bitcoind.rpcuser="
+
+# We are using `exec` to enable gracefull shutdown of running daemon.
+# Check http://veithen.github.io/2014/11/16/sigterm-propagation.html.
+exec btcd $EXTERNAL_IP_OPT \
+--rpcuser=$BITCOIN_NEUTRINO_RPC_USER
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/bitcoin-neutrino/entrypoint.sh]: We decided to open RPC port on neutrino btcd instance?

File: docker/mainnet/bitcoin/bitcoin.mainnet.conf [hunk 3]
```diff
@@ -0,0 +1,18 @@
+port=8333
+
+rpcbind=0.0.0.0
+rpcport=8332
+
+rpcallowip=172.100.2.100
+rpcallowip=172.100.2.101
+
+txindex=1
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/bitcoin/bitcoin.mainnet.conf]: Should we need to add comment here that txindex is needed for lnd?

File: docker/mainnet/connector/connector.mainnet.conf [hunk 4]
```diff
@@ -0,0 +1,54 @@
+[Application Options]
+datadir=/root/.connector
+debuglevel=debug
+network=mainnet
+# RPC address to bind
+rpchost=0.0.0.0
+rpcport=9002
+# exchange address
+enginehost=mainnet.exchange.bitlum.io
+engineport=8080
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: Do we opened this port only for connector?

File: docker/mainnet/connector/connector.mainnet.conf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+bitcoin.host=bitcoin.mainnet
+bitcoin.port=8332
+bitcoin.feeperunit=87
+
+[Bitcoincash]
+bitcoincash.minconfirmations=1
+bitcoincash.syncdelay=5
+bitcoincash.host=bitcoin-cash.mainnet
+bitcoincash.port=9332
+bitcoincash.feeperunit=350
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: We need to change this accordingly with real fee per unit and later make this dynamic

File: docker/mainnet/connector/connector.mainnet.conf [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+[Dash]
+dash.minconfirmations=1
+dash.syncdelay=5
+dash.host=dash.mainnet
+dash.port=10332
+dash.feeperunit=350
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: We need to change this accordingly with real fee per unit and later make this dynamic

File: docker/mainnet/connector/connector.mainnet.conf [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+ethereum.syncdelay=5
+ethereum.host=ethereum.mainnet
+ethereum.port=11332
+
+[Litecoin]
+litecoin.minconfirmations=1
+litecoin.syncdelay=5
+litecoin.host=litecoin.mainnet
+litecoin.port=12332
+litecoin.feeperunit=350
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: We need to change this accordingly with real fee per unit and later make this dynamic

File: docker/mainnet/connector/connector.mainnet.conf [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+# exchange address
+enginehost=mainnet.exchange.bitlum.io
+engineport=8080
+
+[Bitcoin]
+bitcoin.minconfirmations=1
+bitcoin.syncdelay=5
+bitcoin.host=bitcoin.mainnet
+bitcoin.port=8332
+bitcoin.feeperunit=87
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: We need to change this accordingly with real fee per unit and later make this dynamic. In this case unit is not "
"byte", but "vbyte".

File: docker/mainnet/connector/connector.mainnet.conf [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+litecoin.feeperunit=350
+
+[Bitcoinlightning]
+bitcoinlightning.tlscertpath=/root/.lnd/tls.cert
+bitcoinlightning.macaroonpath=/root/.lnd/admin.macaroon
+# lnd RPC address
+bitcoinlightning.host=bitcoin-lightning.mainnet
+bitcoinlightning.port=10009
+# lnd P2P address
+bitcoinlightning.peerhost=connector.bitlum.io
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/connector/connector.mainnet.conf]: Do we have this domain name created?

File: docker/mainnet/.env.empty (hunk 1 above)
Comment (someanon) [reply docker/mainnet/.env.empty]: sure, will fix

File: docker/mainnet/dash/dash.mainnet.conf [hunk 10]
```diff
@@ -0,0 +1,11 @@
+port=10333
+rpcport=10332
+
+rpcallowip=172.100.2.100
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/dash/dash.mainnet.conf]: Should we put comment who is allowed to make rpc call here?

File: docker/mainnet/dash/dash.mainnet.conf [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+txindex=1
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/dash/dash.mainnet.conf]: Why do we need txindex in dash?

File: docker/mainnet/litecoin/litecoin.mainnet.conf [hunk 12]
```diff
@@ -0,0 +1,16 @@
+port=12333
+
+rpcbind=0.0.0.0
+rpcport=12332
+
+rpcallowip=172.100.2.100
+
+addresstype=legacy
+
+txindex=1
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/litecoin/litecoin.mainnet.conf]: Why do we need txindex in litecoin?

File: docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf [hunk 13]
```diff
@@ -0,0 +1,15 @@
+port=9333
+
+rpcbind=0.0.0.0
+rpcport=9332
+
+rpcallowip=172.100.2.100
+
+txindex=1
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf]: Why do we need txindex in dash?

File: docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+usecashaddr=0
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf]: Is that somehow affects type of addresses user could use?

File: docker/mainnet/ethereum/ethereum.mainnet.conf [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+Percentile = 60
+
+[Shh]
+MaxMessageSize = 1048576
+MinimumAcceptedPOW = 2e-01
+
+[Node]
+NoUSB = true
+DataDir = "/root/.ethereum"
+IPCPath = "/root/.ethereum/geth.ipc"
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/ethereum/ethereum.mainnet.conf]: Why do we need ipc enabled?

File: docker/mainnet/ethereum/ethereum.mainnet.conf [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+HTTPHost = "0.0.0.0"
+HTTPPort = 11332
+HTTPModules = ["net", "web3", "eth", "shh"]
+HTTPVirtualHosts = ["ethereum.mainnet"]
+WSHost = "0.0.0.0"
```
Comment (AndrewSamokhvalov) [inline docker/mainnet/ethereum/ethereum.mainnet.conf]: Why do we need websocket enabled?

File: docker/mainnet/bitcoin-neutrino/entrypoint.sh (hunk 2 above)
Comment (someanon) [reply docker/mainnet/bitcoin-neutrino/entrypoint.sh]: No, it closed.

File: docker/mainnet/connector/connector.mainnet.conf (hunk 4 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: We do not have our exchange in mainnet and we run with `--disableexchange` option. We can remove this option but i'm not sure if it isn't mandatory or something.

File: docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf (hunk 14 above)
Comment (someanon) [reply docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf]: it shouldn't, it is just about generating new address on local wallet. But we still have problems with new address types validation.

File: docker/mainnet/ethereum/ethereum.mainnet.conf (hunk 15 above)
Comment (someanon) [reply docker/mainnet/ethereum/ethereum.mainnet.conf]: It's default option generated by `geth dumpconfig`.

File: docker/mainnet/ethereum/ethereum.mainnet.conf (hunk 16 above)
Comment (someanon) [reply docker/mainnet/ethereum/ethereum.mainnet.conf]: Same as IPC - default option.

File: docker/mainnet/litecoin/litecoin.mainnet.conf (hunk 12 above)
Comment (someanon) [reply docker/mainnet/litecoin/litecoin.mainnet.conf]: Not sure, but it was in testnet and simnet in previous installations

File: docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf (hunk 13 above)
Comment (someanon) [reply docker/mainnet/bitcoin-cash/bitcoin-cash.mainnet.conf]: Not sure, but it was in testnet and simnet in previous installations

File: docker/mainnet/connector/connector.mainnet.conf (hunk 5 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: What is our real `fee per unit`?

File: docker/mainnet/connector/connector.mainnet.conf (hunk 6 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: What is our real `fee per unit`?

File: docker/mainnet/connector/connector.mainnet.conf (hunk 7 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: What is our real `fee per unit`?

File: docker/mainnet/connector/connector.mainnet.conf (hunk 8 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: What is our real `fee per unit`?

File: docker/mainnet/connector/connector.mainnet.conf (hunk 9 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: Not yet.

File: docker/mainnet/dash/dash.mainnet.conf (hunk 10 above)
Comment (someanon) [reply docker/mainnet/dash/dash.mainnet.conf]: Yes, nice idea.

File: docker/mainnet/dash/dash.mainnet.conf (hunk 11 above)
Comment (someanon) [reply docker/mainnet/dash/dash.mainnet.conf]: Not sure, but it was in testnet and simnet in previous installations

File: docker/mainnet/bitcoin/bitcoin.mainnet.conf (hunk 3 above)
Comment (someanon) [reply docker/mainnet/bitcoin/bitcoin.mainnet.conf]: t required by lnd. We can add comments about this txindex.

File: docker/mainnet/bitcoin-neutrino/entrypoint.sh (hunk 2 above)
Comment (someanon) [reply docker/mainnet/bitcoin-neutrino/entrypoint.sh]: Just looked, it opened to the private network

Comment (someanon) [reply docker/mainnet/bitcoin-neutrino/entrypoint.sh]: It was openned by docker compose, but not by Dockerfile or by btcd config. I will leave this `--rpcuser` and `--rpcpass` options for security reasons and RPC still would be accessible inside container.

File: docker/mainnet/connector/connector.mainnet.conf (hunk 9 above)
Comment (someanon) [reply docker/mainnet/connector/connector.mainnet.conf]: Will add this domain when will start to test mainnet.

File: docker/mainnet/ethereum/ethereum.mainnet.conf (hunk 15 above)
Comment (someanon) [reply docker/mainnet/ethereum/ethereum.mainnet.conf]: Disabled IPC and WS

File: docker/mainnet/ethereum/ethereum.mainnet.conf (hunk 16 above)
Comment (someanon) [reply docker/mainnet/ethereum/ethereum.mainnet.conf]: Disabled IPC and WS

**Pull request merged**